generators/chipyard/src/main/scala/TestHarness.scala

```diff
@@ -39,7 +42,9 @@ class TestHarness(implicit val p: Parameters) extends Module with HasHarnessSign
   // dutReset assignment can be overridden via a harnessFunction, but by default it is just reset
   val dutReset = WireDefault(if (p(GlobalResetSchemeKey).pinIsAsync) reset.asAsyncReset else reset)
 
-  ldut.harnessFunctions.foreach(_(this))
-
+  ldut match { case d: HasTestHarnessFunctions =>
```

Comment:
A nit: Since we now call `lSystem` -> `lazySystem`, can we do the same to the DUT.